File: be_hotel/src/main/java/no/services/account/AccountService.java
package no.services.account;

import no.modal.entity.Account;
import no.modal.reponse.AuthReponse;
import no.modal.request.AccountRequest;
import org.springframework.data.domain.Page;

public interface AccountService {
    // get
    Page<Account> findAll(int page, int size);
    Account findById(long id);
    // create
    Account createAccount(AccountRequest form);
    // update
    Account updateAccount(AccountRequest form, long id);
    // delete
    boolean deleteAccount(long id);
    // login
    AuthReponse login(String username, String password);
}

File: fe_hotel/web_app/src/services/AccountService.js
import axiosInstance from "./axiosConfigs"

const AccountService = {
    create(body){
        return axiosInstance.post('Account/createAccount', body)
    },
    getAll(page, size){
        return axiosInstance.get(`Account/getAllAccount/${page}/${size}`)
    },
    getId(id){
        return axiosInstance.get(`Account/getById/${id}`)
    },
    update(id, body){
        return axiosInstance.put(`Account/updateAccount/${id}`)
    },
    delete(id){
        return axiosInstance.get(`Account/deleteAccount/${id}`)
    }
}
export default AccountService;